cpp/src/examples/gcylon/gsorting_plus.cpp

```diff
@@ -0,0 +1,226 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Sorting a cudf table vs merging multiple sorted tables
+ * both with the same number of rows
+ */
+
+#include <iostream>
+#include <fstream>
+#include <string>
+
+#include <glog/logging.h>
+
+#include <cudf/table/table.hpp>
+#include <cudf/io/csv.hpp>
+#include <cudf/sorting.hpp>
+#include <cudf/merge.hpp>
+#include <cudf/quantiles.hpp>
+#include <cuda.h>
+
+#include <gcylon/utils/construct.hpp>
+#include <examples/gcylon/print.hpp>
+
+using namespace gcylon;
+using namespace std::chrono;
+
+int64_t calculateRows(std::string dataSize, const int& cols) {
```

Comment:
```suggestion
int64_t calculateRows(const std::string& dataSize, int cols) {
```
Otherwise the string would get copied here. and its efficient to copy primitive types to methods rather than using their references.